tests/integration/veth_vlan_vxlan.yaml

```diff
@@ -0,0 +1,370 @@
+description: Remove the vlan interface eth1.101 and eth1.102
+interfaces:
+- name: eth1.101
+  type: vlan
+  state: absent
+- name: eth1.102
+  type: vlan
+  state: absent
+description: Create VLAN with vlan ID 101 and vlan ID 102 on eth1
+interfaces:
+- name: eth1.101
+  type: vlan
+  state: up
+  vlan:
+    base-iface: eth1
+    id: 101
+    reorder-headers: true
+- name: eth1.102
+  type: vlan
+  state: up
+  vlan:
+    base-iface: eth1
+    id: 102
+    reorder-headers: true
+description: Bring up the dummy device dummy00, configure the vlan dummy00.101 over dummy00 with ID 101
+interfaces:
+- name: dummy00
+  type: dummy
+  state: up
+- name: dummy00.101
+  type: vlan
+  state: up
+  vlan:
+    base-iface: dummy00
+    id: 101
+    reorder-headers: true
+description: Delete the dummy device dummy00 and delete the vlan device dummy00.101
+interfaces:
+- name: dummy00
+  type: dummy
+  state: absent
+- name: dummy00.101
+  type: vlan
+  state: absent
+description: Create bond interface bond0 with bonding mode balance-rr and port eth1 attached, create the vxlan interface bond0.201 with vxlan ID 201, remote set to 192.168.100.2, learning set to true, and the destination-port set to 4789
+interfaces:
+- name: bond0
+  type: bond
+  state: up
+  link-aggregation:
+    mode: balance-rr
+    port:
+    - eth1
+- name: bond0.201
+  type: vxlan
+  state: up
+  vxlan:
+    base-iface: bond0
+    id: 201
+    learning: true
+    remote: 192.168.100.2
+    destination-port: 4789
+description: Delete the both ovs bridge ovs-br0 and linux bridge br0
+interfaces:
+- name: br0
+  type: linux-bridge
+  state: absent
+- name: ovs-br0
+  type: ovs-bridge
+  state: absent
+description: Configure the veth device veth1 and its peer device veth1peer
+interfaces:
+- name: veth1
+  type: ethernet
+  state: up
+  veth:
+    peer: veth1peer
+- name: veth1peer
+  type: ethernet
+  state: up
+description: Delete the both veth1 and veth1peer devices
+interfaces:
+- name: veth1
+  type: ethernet
+  state: absent
+- name: veth1peer
+  type: ethernet
+  state: absent
+description: Delete the veth device veth1 and the veth peer device veth1peer
+interfaces:
+- name: veth1
+  type: ethernet
+  state: absent
+- name: veth1peer
+  type: ethernet
+  state: absent
+description: Configure the veth veth1 to have the peer veth1peer, configure the veth veth1peer to have the peer veth1
+interfaces:
+- name: veth1
+  type: ethernet
+  state: up
+  veth:
+    peer: veth1peer
+- name: veth1peer
+  type: ethernet
+  state: up
+  veth:
+    peer: veth1
+description: Set the veth1 peer device to be veth2peer
```

Comment:
The YAML is mark veth1 and veth1peer as absent.

Reply:
This is strange that I only set the description in one place: 

```
    @pytest.mark.tier1
    def test_modify_veth_peer(self):
        with veth_interface(VETH1, VETH1PEER) as d_state:
            d_state[Interface.KEY][0][Veth.CONFIG_SUBTREE][
                Veth.PEER
            ] = VETH2PEER
            apply_with_description(
                "Set the veth1 peer device to be veth2peer", d_state
            )

            c_state = statelib.show_only(
                (
                    VETH1,
                    VETH2PEER,
                )
            )
            assert (
                c_state[Interface.KEY][0][Veth.CONFIG_SUBTREE][Veth.PEER]
                == VETH2PEER
            )
            assert c_state[Interface.KEY][1][Interface.NAME] == VETH2PEER
            ```
            
            However, for the same description, it corresponds to multiple states, could this be a bug somewhere: 
            
            ```
description: Set the veth1 peer device to be veth2peer
interfaces:
- name: veth1
  type: ethernet
  state: absent
- name: veth1peer
  type: ethernet
  state: absent 
description: Set the veth1 peer device to be veth2peer
interfaces:
- name: veth1
  type: ethernet
  state: up
  veth:
    peer: veth2peer            
```

Reply:
Because `apply_with_description` modified input `desired_state`.

Next time, please investigate it by yourself.

Reply:
https://github.com/nmstate/nmstate/pull/2750

Reply:
Thanks, I updated the PR.